src/pages/workspace/WorkspaceEditorPage.js

```diff
@@ -0,0 +1,154 @@
+        this.uploadAvatarPromise = Promise.resolve();
+    }
+
+    onImageSelected(image) {
+        this.setState({previewAvatarURL: image.uri});
+
+        // Store the upload avatar promise so we can wait for it to finish before updating the policy
+        this.uploadAvatarPromise = uploadAvatar(image).then(url => new Promise((resolve) => {
+            this.setState({avatarURL: url}, resolve);
+        }));
+    }
+
+    onImageRemoved() {
+        this.setState({previewAvatarURL: '', avatarURL: ''});
+    }
+
+    submit() {
+        // Wait for the upload avatar promise to finish before updating the policy
+        this.uploadAvatarPromise.then(() => {
+            const name = this.state.name.trim();
+            const avatarURL = this.state.avatarURL;
+            const policyID = this.props.policy.id;
+
+            update(policyID, {name, avatarURL});
+        });
+    }
+
+    render() {
+        const {policy} = this.props;
+
+        if (!Permissions.canUseFreePlan(this.props.betas)) {
+            console.debug('Not showing workspace editor page because user is not on free plan beta');
+            return <Navigation.DismissModal />;
+        }
+
+        if (_.isEmpty(policy)) {
+            return null;
+        }
+
+        return (
+            <ScreenWrapper>
+                <HeaderWithCloseButton
+                    title={this.props.translate('workspace.editor.title')}
+                    onCloseButtonPress={Navigation.dismissModal}
+                />
+
+                <View style={[styles.pageWrapper, styles.flex1, styles.pRelative]}>
+                    <View style={[styles.w100, styles.flex1]}>
+                        <AvatarWithImagePicker
+                            avatarURL={this.state.previewAvatarURL}
+                            DefaultAvatar={() => (
+                                <Icon
+                                    src={Workspace}
+                                    height={80}
+                                    width={80}
+                                    fill={defaultTheme.icon}
+                                />
+                            )}
+                            style={[styles.mb3]}
+                            anchorPosition={{top: 172, right: 18}}
+                            isUsingDefaultAvatar={!this.state.previewAvatarURL}
+                            onImageSelected={this.onImageSelected}
+                            onImageRemoved={this.onImageRemoved}
+                        />
+
+                        <TextInputWithLabel
+                            label={this.props.translate('workspace.editor.nameInputLabel')}
+                            value={this.state.name}
+                            onChangeText={name => this.setState({name})}
+                            onSubmitEditting={this.submit}
+                        />
+                        <Text style={[styles.mt2]}>{this.props.translate('workspace.editor.nameInputHelpText')}</Text>
```

Comment:
Can we also use `fontSizeLabel` and `colorMuted` here? This way it appears like hint text under the input.

Reply:
I see there's a formHint style that has those! I just sent the commit with the changes 😄